pkg/apis/clusteroperator/types.go

```diff
@@ -384,6 +384,9 @@ type MachineSetSpec struct {
 	// MachineSetConfig is the configuration for the MachineSet
 	MachineSetConfig
 
+	// ClusterHardware specifies the hardware that the cluster will run on
+	ClusterHardware ClusterHardwareSpec
```

Comment:
Do we foresee having other settings from the cluster (that are not part of the hardware spec) that should also be included in the machineset?

Reply:
I don't know. Currently, the only thing in `ClusterSpec` that is not passed along to `MachineSetSpec` in some way or another is the `ClusterConfigSpec`. It shouldn't be too difficult to adjust what we pass along to `MachineSetSpec` in the future if we need something more. Are you suggesting that it might be better to have an explicit struct containing all of the information passed along from the cluster rather than using the `ClusterHardwareSpec` at the top-level in the `MachineSetSpec`?

Reply:
> Are you suggesting that it might be better to have an explicit struct containing all of the information passed along from the cluster rather than using the ClusterHardwareSpec at the top-level in the MachineSetSpec?

Not sure that's better. So, no :) 
I guess we can add other fields to the machine set spec as needed if we need something else from the cluster.